docs/portal.properties-Reference.md

```diff
@@ -280,3 +281,58 @@ This gene set will add the following in the query box:
 ```
 "BRCA genes" BRCA1: MUT=E1258D; BRCA2: HOMDEL MUT=NONSENSE MUT=NONSTART MUT=NONSTOP MUT=FRAMESHIFT MUT=SPLICE MUT=TRUNC;
 ```
+# Ehcache Settings
+cBioPortal is supported on the backend with Ehcache. The configuration, size, and location of these caches are configurable from within portal.properties through the following properties.
+
+First, select a cache configuration using `ehcache.xml_configuration`. This should be the name of an Ehcache xml configuration file; the default provided is `ehcache.xml` which configures a heap-only cache. To change caching configuration, directly edit `/ehcache.xml`. Alternatively, you can create your own Ehcache xml configuration file, place it under `/persistence/persistence-api/src/main/resources/` and set `ehcache.xml_configuration` to `/<Ehcache xml configuration filename>`.  
+```
+ehcache.xml_configuration=
+```
+
+If the cache is configured to use disk resources, users must make a directory available and set it with the `ehcache.persistence_path` property. Ehcache will create separate directories under the provided path for each cache defined in the ehcache.xml_configuration file. 
+```
+ehcache.persistence_path=[location on the disk filesystem where Ehcache can write the cache to /tmp/]
+```
+
+Cache size must be set for heap and/or disk depending which are in use; Ehcache requires disk size to be greater than heap size in a hybrid configuration. Zero is not a supported size and will cause an exception. Units are in megabytes. Default values are provided. The general repository cache is specified to use 5120MB of heap. The static repository cache is specified to use 30MB of heap. For installations with increased traffic or data, cache sizes can be increased to further improve performance. 
```

Comment:
Nice description! Only minor comment is that 5120MB of heap might be quite big - our current hardware requirements say a minimum of 2GB of RAM is required to run the portal. Maybe the default should be to not use any caching? Like only use if you opt in and know what you're doing kind of thing?

Reply:
We handled this by having Ehcache be disabled with the default configuration. Now if users want to use caching, the set the property in portal.properties and uncomment the lines setting disk / heap usage. (and maybe adjust)